src/storage-controller/src/lib.rs

```diff
@@ -718,19 +730,31 @@ where
 
             // Install the collection state in the appropriate spot.
             match &collection_state.data_source {
-                DataSource::Introspection(_) => {
+                DataSource::Introspection(typ) => {
                     debug!(data_source = ?collection_state.data_source, meta = ?metadata, "registering {} with persist monotonic worker", id);
-                    self.persist_monotonic_worker.register(id, write);
+                    // We always register the collection with the collection manager,
+                    // regardless of read-only mode. The CollectionManager itself is
+                    // aware of read-only mode and will not attempt to write before told
+                    // to do so.
+                    //
+                    self.register_introspection_collection(id, *typ, write)
```

Comment:
Is it safe to pull this call up here? It's hard to tell what in the method is a delicate dance of doing things in _just_ the right order and what's historical cruft